render/wlr_texture.c

```diff
@@ -22,6 +22,10 @@ void wlr_texture_destroy(struct wlr_texture *texture) {
 struct wlr_texture *wlr_texture_from_pixels(struct wlr_renderer *renderer,
 		uint32_t fmt, uint32_t stride, uint32_t width, uint32_t height,
 		const void *data) {
+	assert(width > 0);
+	assert(height > 0);
+	assert(stride >= width);
```

Comment:
Now that I think of it, the stride can be smaller than the width for formats with a number of bits per pixel lower than 8. We don't support any of these yet.